### ISSUE TITLE
API rate limit exceeded

### ISSUE THREAD

**Issue by marekmatula** [clockfort/GitHub-Backup#42]
Hi 
I tried to download a full backup of my GitHub account.

` github-backup -v all -a owner --all -p token  username ./`

The script is not able to download issues and fails with the API rate limit exceeded. See traceback below.
I repeat it after more than one hour. It fails with the same error.

I found this issue which can be useful for a fix: https://github.com/PyGithub/PyGithub/issues/1319

```
Traceback (most recent call last):
  File "/home/mm/.local/bin/github-backup", line 8, in <module>
    sys.exit(main())
  File "/home/mm/.local/lib/python3.8/site-packages/github_backup/github_backup.py", line 133, in main
    process_account(gh, account, args)
  File "/home/mm/.local/lib/python3.8/site-packages/github_backup/github_backup.py", line 305, in process_account
    RepositoryBackup._backup_issues(issues, args, dir)
  File "/home/mm/.local/lib/python3.8/site-packages/github_backup/github_backup.py", line 435, in _backup_issues
    issue_data.setdefault('event_data', []).append(event.raw_data)
  File "/home/mm/.local/lib/python3.8/site-packages/github/GithubObject.py", line 103, in raw_data
    self._completeIfNeeded()
  File "/home/mm/.local/lib/python3.8/site-packages/github/GithubObject.py", line 298, in _completeIfNeeded
    self.__complete()
  File "/home/mm/.local/lib/python3.8/site-packages/github/GithubObject.py", line 305, in __complete
    headers, data = self._requester.requestJsonAndCheck("GET", self._url.value)
  File "/home/mm/.local/lib/python3.8/site-packages/github/Requester.py", line 315, in requestJsonAndCheck
    return self.__check(
  File "/home/mm/.local/lib/python3.8/site-packages/github/Requester.py", line 340, in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.RateLimitExceededException: 403 {"message": "API rate limit exceeded for user ID 13673271.", "documentation_url": "https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
```

**Comment by mendhak** [comment]
I've added a rate limit retry ability in my fork, you can [see it here](https://github.com/mendhak/GitHub-Backup/commit/ec9a41f545ac3d669ee22bf66ed760285e84d63c#diff-d670a69743a94edda5357cbb35836d8606570b6a1c60ec3dddfb8881da1b59fbR155).  

Also I've submitted a [pull request](https://github.com/clockfort/GitHub-Backup/pull/44).  It's not a great looking approach, it basically involves wrapping each method, in a method, with a decorator, but seems to work well for me.  

If you want to try it out, use

    pip install git+https://github.com/mendhak/GitHub-Backup

**Comment by stv0g** [comment]
This is now hopefully fixed by #42

**Comment by hellkite500** [comment]
Looks like PyGithub/PyGithub#2387 will make it possible to handle this directly in the library.
